src/main/java/org/symphonyoss/symphony/messageml/elements/TableSelect.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2016-2017 MessageML - Symphony LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.symphonyoss.symphony.messageml.elements;
+
+import org.commonmark.node.Node;
+import org.symphonyoss.symphony.messageml.exceptions.InvalidInputException;
+import org.symphonyoss.symphony.messageml.markdown.nodes.form.TableSelectNode;
+import org.symphonyoss.symphony.messageml.util.XmlPrintStream;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Class representing a TableSelect inside a Form.
+ *
+ * @author Pedro Sanchez
+ * @since 05/30/2019
+ */
+public class TableSelect extends FormElement {
+
+  public static final String MESSAGEML_TAG = "tableselect";
+
+  private static final String NAME_ATTR = "name";
+  private static final String TYPE_ATTR = "type";
+  private static final String HEADER_TEXT_ATTR = "header-text";
+  private static final String BUTTON_TEXT_ATTR = "button-text";
+  private static final String POSITION_ATTR = "position";
+
+  private static final String PRESENTATIONML_TABLE_TAG = "table";
+
+  public static final Set<String> VALID_TYPES = new HashSet<>(Arrays.asList("button", "checkbox"));
+
+  public static final Set<String> VALID_POSITIONS = new HashSet<>(Arrays.asList("left", "right"));
+
+  private final static String LEFT = "left";
+  private final static String RIGHT = "right";
+  private final static String BUTTON = "button";
+  private final static String CHECKBOX = "checkbox";
+
+  public TableSelect(Element parent) {
+    super(parent, MESSAGEML_TAG);
+
+    setAttribute(HEADER_TEXT_ATTR, "Select");
+    setAttribute(BUTTON_TEXT_ATTR, "SELECT");
+  }
+
+
+  @Override
+  public void validate() throws InvalidInputException {
+    super.validate();
+
+    String name = getAttribute(NAME_ATTR);
+
+    if (name == null || name.trim().isEmpty()) {
+      throw new InvalidInputException("The attribute \"name\" is required");
+    }
+
+    String type = getAttribute(TYPE_ATTR);
```

Comment:
Why are you assigning this to a variable instead of using the function directly? You're not going to use the value for anything but the validation anyway. Same for other places doing the same thing.

Reply:
Refactored to use the assertRequired